token/program-2022/src/state.rs

```diff
@@ -80,6 +103,55 @@ impl Pack for Mint {
     }
 }
 
+/// MintWithTransferFee data.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct MintWithTransferFee {
+    /// Base mint data, done as composition
+    pub base_mint: Mint,
+
+    /// ALL NEW DATA MUST START FROM HERE
+    /// account type, always set to "MintWithTransferFee"
+    pub account_type: AccountType,
+    /// Any provided fee account must be owned by this key. If set to `None`, then
+    /// this mint behaves like a normal `Mint`.
+    pub fee_account_owner: COption<Pubkey>,
```

Comment:
Maybe there's two authorities here?
* `fee_withdraw_authority` - who can get at the `withheld_amount` fields
* `fee_config_authority` - who can change the fee

Reply:
Yep, that makes more sense